### PR TITLE
[PR] Implement outdoor POI search functionality and map integration (ES-5)

### LINKED ISSUE #46: [ES-5] Points of Interest
This epic enables users to discover outdoor points of interest around campus and navigate to them.

### LINKED ISSUE #47: [US-5.1] Find nearby outdoor points of interest
As a user, I want to find the nearest outdoor point of interest by type or range, so that I can quickly locate useful services.

Acceptance Criteria:
- [x] POIs can be filtered by type and range.
- [x] Results are sorted by distance ( we should only show POIs 10KM around the campus)

### LINKED ISSUE #48: [US-5.2] Get directions to an outdoor point of interest
As a user, I want directions to a selected outdoor point of interest, so that I can reach it easily.

Acceptance Criteria:
- [ ] A route action is available for POIs.
- [ ] Directions are displayed on the map.
- [ ] Switching POIs updates the route.

### LINKED ISSUE #285: [T-5.1.1] Find nearby outdoor POIs
As a user, I want to find the nearest outdoor point of interest by type or range, so that I can quickly locate useful services.

Acceptance Criteria:
- [ ] POIs can be filtered by type and range.
- [ ] Results are sorted by distance?

### LINKED ISSUE #286: [T-5.2.1] Get directions to an outdoor point of interest
As a user, I want directions to a selected outdoor point of interest, so that I can reach it easily.

Acceptance Criteria:
- [ ] A route action is available for POIs.
- [ ] Directions are displayed on the map.
- [ ] Switching POIs updates the route.

### LINKED ISSUE #311: [REFACTOR] Refactor Outdoor POI Search To Use Reactive User Location State
🛠️ **Refactoring Type**  
State management refactor

🧾 **Reason**  
useOutdoorPOI was receiving userPositionRef.current from MapScreen, but ref updates do not trigger rerenders. That could leave outdoor POI searches using a stale or fallback location even after the real user location was received.

🔗 **Related PR/Branch**  
feature/ES-5

📂 **Affected Area**  
mobile-app/components/MapScreen.tsx

📌 **Resolution Note**  
When this refactoring task is completed and committed, please comment the commit hash/number on this issue itself.

### PR DESCRIPTION
**Pull request by DaniK1001**
# Notes:
- This PR closes #46 
and all its sub-issues:
  - closes #47 
    - closes #285 
  - closes #48 
    - closes #286  
 
- This Epic was implemented by one person due to its issues being closely related. hence, we also decided to make one PR for the whole feature 

Acceptance Criteria:
- [x] A route action is available for POIs.
- [x] Directions are displayed on the map.
- [x] Switching POIs updates the route.
- [x] POIs can be filtered by type and range.
- [x] Results are sorted by distance ( we should only show POIs 10KM around the campus)
  
---

# Outdoor Points of Interest (POI) Feature (Last feature! 🎉)
 
## Overview
 
This PR introduces an Outdoor POIs feature that lets users discover nearby points of interest (restaurants, cafes, pharmacies, etc.) around Concordia's SGW and Loyola campuses, view details about them, and navigate to them — along with comprehensive unit tests covering the new services, components, and navigation logic.
 
---
 
### Default (no search)
 
Google Maps' native POIs remain visible on the map as usual. Tapping any native POI opens an info card showing the name, address (resolved asynchronously via `expo-location`'s `reverseGeocodeAsync` — no API key required), category, and a Directions button. Pressing Directions starts coordinate-based navigation to that POI; it is treated as a generic destination, not a campus building.
 
### Search
 
When a user types a supported category (e.g. `restaurant`, `cafe`, `pharmacy`) in the search bar:
 
- A "Searching nearby places…" loading indicator appears in the dropdown while results are fetched. The keyboard does not auto-dismiss on submit, keeping the dropdown visible.
- Results appear with name, address, and category.
- On selection or Enter, Google's native POIs are hidden and replaced with custom markers — 32px Concordia-red circles with a white border and a category-specific icon (fork/knife for restaurants, coffee cup for cafes, etc.). Markers render above building polygons via a high z-index, so tapping a marker on a building registers as a POI tap.
- Clearing the search bar removes custom markers and restores native POIs.
 
### Search Radius & Centering
 

- Results are fetched within a 1km radius centered on the user's GPS location (`Location.getCurrentPositionAsync`), falling back to the active campus center (SGW or Loyola) if GPS is unavailable. The Google Places API is called with `rankby=distance`; a client-side filter enforces the 1km cap.
- Results are ranked by distance, so users see the closest matches first. By design, we currently consume only the first Google Places Nearby Search page (up to 20 results) and do not paginate with next_page_token. This is an intentional strategy to keep the map readable (reduce marker clutter around campus) and to control Places API usage/quota by avoiding unnecessary follow-up requests.

 
### Supported Categories
 
| Search terms | Google Places type |
|---|---|
| `restaurant`, `restaurants`, `food` | `restaurant` |
| `cafe`, `coffee`, `coffeeshop` | `cafe` |
| `pharmacy`, `drugstore` | `pharmacy` |
| `gym` | `gym` |
| `bank` | `bank` |
| `supermarket`, `grocery` | `supermarket` |
| `bar` | `bar` |
| `hospital` | `hospital` |
| `library` | `library` |
| `parking` | `parking` |
| `gas`, `gas_station` | `gas_station` |
| `hotel` | `hotel` |
 
All terms are case-insensitive and whitespace-trimmed.
 
### POI Info Card
 
Displays name, address, category badge, rating (if available), and open/closed status (if available). Has a close button and tappable backdrop to dismiss, plus a Directions button that triggers `openNavigationForCoordinate(label, coordinate)` — a coordinate-based path that does not treat the destination as a campus building.
 
---
 
## Architecture
 
- `outdoorPOIService.ts` — category resolution (`isSupportedPOICategory`), Google Places Nearby Search (`fetchNearbyPOIs`), Place Details fetch (`fetchPlaceDetails`), campus center helper.
- `useOutdoorPOI.ts` — hook managing POI state: fetching on search, selection, address resolution via `reverseGeocodeAsync`, GPS position resolution.
- `OutdoorPOIInfoCard.tsx` — info card component for selected POIs.
- `MapScreen.tsx` — integrates everything: merged search results, marker rendering, `onPoiClick` for native POIs, map style toggling, directions flow.
- `useNavigationBetweenBuildings.ts` — extended with `openNavigationForCoordinate` for coordinate-based POI navigation.
 
Outdoor POIs are fully decoupled from buildings, indoor POIs, and `useSelectedBuilding`. Building taps only register when the user taps directly on a building's polygon — no snap-to-nearest-building behavior.
 
---
 
### `outdoorPOIService.test.ts`
 
Covers category resolution (`isSupportedPOICategory`), fetching nearby POIs from the Google Maps API (`fetchNearbyPOIs`), fetching POI details (`fetchPlaceDetails`), and campus center logic (`getCampusCenterForPOI`), including API integration, error handling, and edge cases.
 
### `OutdoorPOIInfoCard.test.tsx`
 
Verifies correct rendering of POI details, handling of missing data, open/closed status display, category formatting, color-blind mode, and button interactions for closing and requesting directions.
 
### `useNavigationBetweenBuildings` hook tests (extended)
 
Covers the new `openNavigationForCoordinate` method, validating correct state updates for navigation destination, coordinate, and route type.

### CODE REVIEW

**Review by menaboulus — COMMENTED** [review]
One thing that could be improved is replacing global with globalThis in the fetch mocks to follow best practices and avoid warnings. Also, some mock setup is repeated across tests extracting it into shared helpers could make the tests cleaner and easier to maintain.

**Review by Lemonada10 — APPROVED** [review]

**Review by Anthony4044 — APPROVED** [review]

**Review by VladShteren — APPROVED** [review]
Tested the feature and the core flow works well: nearby outdoor POIs can be searched, displayed on the map, selected, and used for navigation.

**Comment by MassPaol** [comment]
Thank you @DaniK1001 @leonlolleonlol @VladShteren for completing all my refactoring requests. I reviewed them and they seem to be good so nice job! Ill accept the PR now.

**Review by MassPaol — APPROVED** [review]

File: mobile-app/components/MapScreen.tsx [hunk 1]
```diff
@@ -1801,30 +1877,41 @@ export default function MapScreen() {
...
+          const { placeId, name, coordinate } = e.nativeEvent;
+          if (!coordinate || !name) return;
+          handleCloseCard();
+          selectPOIFromMap({
+            id: placeId ?? `map-poi-${Date.now()}`,
+            name,
+            address: '',
+            coordinate: { latitude: coordinate.latitude, longitude: coordinate.longitude },
+            category: 'place',
+          });
```
[the remark this answers is not in the document]
Comment (DaniK1001) [reply mobile-app/components/MapScreen.tsx]: this could be an enhancement, do you guys think we need it?
When the user taps a Google Maps native POI on the map, your code only saves a very basic POI object, so the info card won’t have extra details like rating or whether it’s open.

**Review by Reuven01 — APPROVED** [review]
All good, guys!! Great Work!! Both the app's functionality and code quality are good!
Thank you

**Review by RyanCheung03 — APPROVED** [review]
App is working well functionally. I looked into some code improvements and didn't find much considering a lot of the refactoring already done! Good job team. Approved

**Review by ammarranko — APPROVED** [review]

**Review by leonlolleonlol — APPROVED** [review]

File: mobile-app/hooks/useOutdoorPOI.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      Location.reverseGeocodeAsync(basePOI.coordinate)
+        .then((results) => {
+          if (results.length === 0) return;
+          const address = results[0].formattedAddress || formatGeocodedAddress(results[0]);
+          if (!address) return;
+          setSelectedOutdoorPOI((prev) => (prev?.id === basePOI.id ? { ...prev, address } : prev));
+        })
+        .catch((error) => {
+          console.warn('useOutdoorPOI: reverse geocoding failed for selected POI', error);
+        });
```
[the remark this answers is not in the document]
Comment (DaniK1001) [reply mobile-app/hooks/useOutdoorPOI.ts]: alright thanks!

**Review by menaboulus — APPROVED** [review]
